chia/wallet/wallet_node.py

```diff
@@ -657,7 +655,10 @@ async def receive_state_from_peer(
         concurrent_tasks_cs_heights: List[uint32] = []
 
         # Ensure the list is sorted
-        items = sorted(items_input, key=last_change_height_cs)
+
+        self.log.info(f"Filtering spam, total items: {len(items_input)}")
+        items = await self.wallet_state_manager.filter_spam(list(sorted(items_input, key=last_change_height_cs)))
```

Comment:
could this be done on the full_node side? ie wallet node provides a min xch amount per coin?

Reply:
yes it could, but we would lose a little bit of the power. For example we can't specify that we only filter if we have more than 200 transctions. We also cannot filter based on CAT asset id, which we plan to do later.

On the other hand, it would help a lot for wallets with many TXs as having any more than 100k creates a massive message which can exceed max message sizes

Reply:
maybe both then :)